modules/gapi/test/cpu/gapi_ocv_stateful_kernel_tests.cpp

```diff
@@ -192,7 +192,7 @@ TEST(StatefulKernel, StateIsAutoResetForNewStream)
     // Compilation & testing
     auto ccomp = c.compileStreaming(cv::compile_args(pkg));
 
-    auto path = findDataFile("cv/video/768x576.avi");
+    auto path = findDataFile("cv/video/768x576.avi", false);
```

Comment:
`false` should not be used for files from `opencv_extra`.
`false` is used for files from external additional **optional** share.

Reply:
Can we add a check for `OPENCV_TEST_DATA_PATH` for emptiness? FAILED tests without `OPENCV_TEST_DATA_PATH` can shock people.

Reply:
> FAILED tests without OPENCV_TEST_DATA_PATH can shock people

This is expected behavior.
Launching tests without required test data doesn't make sense. They should not pass at least.
https://github.com/opencv/opencv/wiki/QA_in_OpenCV

Reply:
@alalek  can you clarify this assertion? i faced with the same behavior in my PR and got several comments.
Could you take a look please?
https://github.com/opencv/opencv/pull/20901/files#diff-df9d13b0d9035428deb2704ecdd722cb4a3193ba653554215768156fffec1f62R43

TO my mind it is expected behavior for:
1) throw SkipExpection if no `OPENCV_TEST_DATA_PATH` is set
2) if it set then `fileData` MUST follows up to correct file. If path is incorrect that it MUST fail test.

The question is: do we need to check `initTestDataPathSilent` before fileData failed - or in another words: is my PR (by link above) behavior correct?

Reply:
> throw SkipExpection if no OPENCV_TEST_DATA_PATH is set

`OPENCV_TEST_DATA_PATH` points to required test data (opencv_extra/testdata).
Test which requires data from opencv_extra must fail in that case.
Passed test (even through skip) is wrong behavior.

Test may be skipped if **optional** test data is missing, e.g. DNN models (it is done because full dataset is large).

---

G-API tests SHOULD NOT use `OPENCV_TEST_DATA_PATH` directly.
G-API tests SHOULD use `OPENCV_DNN_TEST_DATA_PATH` once during initialization (see "dnn" module).